app/bundles/PluginBundle/Entity/IntegrationEntity.php

```diff
@@ -0,0 +1,259 @@
+<?php
+/**
+ * @package     Mautic
+ * @copyright   2014 Mautic Contributors. All rights reserved.
+ * @author      Mautic
+ * @link        http://mautic.org
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\PluginBundle\Entity;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\ORM\Mapping as ORM;
+use Mautic\CoreBundle\Doctrine\Mapping\ClassMetadataBuilder;
+use Mautic\CoreBundle\Entity\CommonEntity;
+
+/**
+ * Class Plugin
+ */
+class IntegrationEntity extends CommonEntity
+{
+
+    /**
+     * @var int
+     */
+    private $id;
+
+    /**
+     * @var string
+     */
+    private $integration;
+
+    /**
+     * @var string
+     */
+    private $integrationEntity;
+
+    /**
+     * @var int
+     */
+    private $integrationEntityId;
+
+    /**
+     * @var \DateTime
+     */
+    private $dateAdded;
+
+    /**
+     * @var \DateTime
+     */
+    private $lastSyncDate;
+
+    /**
+     * @var string
+     */
+    private $internalEntity;
+
+    /**
+     * @var int
+     */
+    private $internalEntityId;
+
+    /**
+     * @var array
+     */
+    private $internal;
+
+    public function __construct ()
+    {
+        $this->internal = new ArrayCollection();
+    }
+
+    /**
+     * @param ORM\ClassMetadata $metadata
+     */
+    public static function loadMetadata (ORM\ClassMetadata $metadata)
+    {
+        $builder = new ClassMetadataBuilder($metadata);
+
+        $builder->setTable('integration_entity')
+            ->setCustomRepositoryClass('Mautic\PluginBundle\Entity\IntegrationEntityRepository');
+
```

Comment:
Let's add some indexes on this table: [integration, integration_entity, internal_entity] and [integration, integration_entity, integration_entity_id] (don't forget the migration)